app/lib/registries/full_topic_taxonomy_registry.rb

```diff
@@ -6,6 +6,17 @@ def [](base_path)
       taxonomy[base_path]
     end
 
+    def find_parent_by_id id
+      topic = taxonomy.find { |_, value|
+        value['content_id'] == id
+      }
+
+      title = topic.nil? ? nil : topic[1]['title']
```

Comment:
maybe use a guard here to avoid the two comparisons, i.e.
```return nil if topic.nil?```

Reply:
We'll be removing this method, it's no longer needed.